app/models/user.rb

```diff
@@ -95,7 +95,15 @@ def active_identities
   end
 
   def active_profile
-    @active_profile ||= profiles.verified.find(&:active?)
+    if defined?(@active_profile)
+      if !@active_profile&.active
```

Comment:
I think this might need to be a two-part statement. If `@active_profile` is `nil`, we'll bust the cache each time and make another query, but it's prob not needed right?
```suggestion
      if @active_profile && !@active_profile.active
```

Reply:
Yes! Good catch.

Reply:
Oh no, there are tests breaking now. It seems like in some of the tests we don't want to cache `nil`. I'm going to try also busting the cache on `User::reload` and see if that helps

Reply:
Looking back at this, the behavior up till now has been to [not cache nil values](https://github.com/18F/identity-idp/blob/main/app/models/user.rb#L98C1-L98C58), so maybe we actually want to preserve that behavior?